sklearn/preprocessing/_data.py

```diff
@@ -2305,6 +2307,8 @@ def _sparse_fit(self, X, random_state):
                 self.quantiles_.append(
                         np.nanpercentile(column_data, references))
         self.quantiles_ = np.transpose(self.quantiles_)
+        self.quantiles_ = np.minimum.accumulate(
+                                        self.quantiles_[::-1])[::-1]
```

Comment:
Thank you for opening this. A quick comment to suggest that `np.maximum.accumulate(self.quantiles_)` also fixes the issue and is more readable.

I initially suggested `np.minimum` because it provided a more satisfying output when I noticed the issue the first time on my own data, but after some more investigation it's not consistent so the `np.maximum` way seems to be a better choice because of readability ?

Reply:
I wonder what's causing the error anyways... 
Maybe, ``np.nanquantile`` is responsible for rounding errors or maybe random samples happen to quantile to the same value. Closer inspection may fix the bug permanently, but I think your fix is the best for now.
Thanks for your suggestion!